packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/TransferDetails/template.success.tsx

```diff
@@ -170,53 +195,74 @@ const Success: React.FC<Props> = props => {
             </Copy>
           </RowCopy>
         )}
+        {(props.fiatCurrency === 'USD' || props.fiatCurrency === 'GBP') &&
+          props.account.agent.account !== '' && (
+            <RowCopy>
+              <div>
+                <Title>
+                  <FormattedMessage
+                    id='modals.simplebuy.transferdetails.account'
+                    defaultMessage='Account Number'
+                  />
+                </Title>
+                <Value data-e2e='sbAccountNumber'>
+                  {props.account.agent.account}
+                </Value>
+              </div>
+              <Copy>
+                <CopyClipboardButton address={props.account.agent.account} />
+              </Copy>
+            </RowCopy>
+          )}
         {props.fiatCurrency === 'GBP' && (
           <RowCopy>
             <div>
               <Title>
                 <FormattedMessage
-                  id='modals.simplebuy.transferdetails.account'
-                  defaultMessage='Account Number'
+                  id='modals.simplebuy.transferdetails.sortcode'
+                  defaultMessage='Sort Code'
                 />
               </Title>
-              <Value data-e2e='sbAccountNumber'>
-                {props.account.agent.account}
-              </Value>
+              <Value data-e2e='sbSortCode'>{props.account.agent.code}</Value>
             </div>
             <Copy>
-              <CopyClipboardButton address={props.account.agent.account} />
+              <CopyClipboardButton address={props.account.agent.code} />
             </Copy>
           </RowCopy>
         )}
-        {props.fiatCurrency === 'GBP' && (
+        {props.fiatCurrency === 'EUR' && (
           <RowCopy>
             <div>
               <Title>
                 <FormattedMessage
-                  id='modals.simplebuy.transferdetails.sortcode'
-                  defaultMessage='Sort Code'
+                  id='modals.simplebuy.transferdetails.swift'
+                  defaultMessage='Bank Code (SWIFT / BIC)'
                 />
               </Title>
-              <Value data-e2e='sbSortCode'>{props.account.agent.code}</Value>
+              <Value data-e2e='sbBankCode'>{props.account.agent.code}</Value>
             </div>
             <Copy>
               <CopyClipboardButton address={props.account.agent.code} />
             </Copy>
           </RowCopy>
         )}
-        {props.fiatCurrency === 'EUR' && (
+        {props.fiatCurrency === 'USD' && (
```

Comment:
I believe if you change this to props.account.currency you will not need `as AgentType`